### ISSUE TITLE
WriteConcern option never passes into MongoDB

### ISSUE THREAD

**Issue by ossinkine** [doctrine/mongodb-odm#2292]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.2.0

#### Summary

I'm sorry in advance if I've just not figured out.
I want to be sure the transaction was processed on majority replicas, I've set up `w: majority` option in `doctrine_mongodb.default_commit_options`, and try to debug where this option passed into MongoDB and did not find any cases. [`DocumentPersister::getWriteOptions`](https://github.com/doctrine/mongodb-odm/blob/a88db3e46d335190ea0395d44d30d580a345141e/lib/Doctrine/ODM/MongoDB/Persisters/DocumentPersister.php#L1491) returns `['w' => 'majority']` and then options passew to [`MongoDB\Collection::updateOne`](https://github.com/mongodb/mongo-php-library/blob/master/src/Collection.php#L1066) which never usew `w` key, it expects `writeConcern` key with object instance of `MongoDB\Driver\WriteConcern`.

#### Current behavior

WriteConcern option never passes into MongoDB.

#### How to reproduce

Just try to pass `w` as write options and check it's never used in `mongodb/mongodb`.

#### Expected behavior

WriteConcern correctly passes info `mongodb/mongodb` and exists in request.

**Comment by alcaeus** [comment]
Now that you mention it...updating the commit options to use `WriteConcern` has completely escaped my mind when updating to the new driver. Marking as a bug and will start fixing soon.

**Comment by ossinkine** [comment]
@alcaeus passing `w` option in `DocumentManager::flush` or specifying in Document annotation also does not work. What is a true way to pass it?

**Comment by alcaeus** [comment]
The true way is to pass a [write concern option](https://www.php.net/manual/en/class.mongodb-driver-writeconcern.php):
```php
$dm->flush(['writeConcern' => new \MongoDB\Driver\WriteConcern('majority')])
```

**Comment by ossinkine** [comment]
Looks like [this method](https://github.com/doctrine/mongodb-odm/blob/a88db3e46d335190ea0395d44d30d580a345141e/lib/Doctrine/ODM/MongoDB/Persisters/DocumentPersister.php#L1491) should be removed or refactored because it works wrong and consfusing

**Comment by alcaeus** [comment]
You've found the source of the problem. Want to create a PR to fix the issue? :)

**Comment by ossinkine** [comment]
Yes I can if you describe the correct behavior in your opinion

**Comment by alcaeus** [comment]
Instead of passing on the `w` key, the method you pointed out should create a `WriteConcern` object and trigger a deprecation to let everyone know that passing a `w` key is no longer supported. There are some places where ODM assumes that as default, so be sure to check those out as well. Please let me know (here or in the doctrine or Symfony-devs slacks) if you run into any problems or have questions.

**Comment by ossinkine** [comment]
@alcaeus Please tell which base branch I should use to fix

**Comment by alcaeus** [comment]
2.2.x is our current branch for bug fixing. Thanks!

**Comment by alcaeus** [comment]
Fixed in #2294.
